src/components/ListNestedComponent.tsx

```diff
@@ -21,6 +22,20 @@ export class ListNestedComponent extends React.Component<any, any> {
         this.props.modelChanged(newModel, success, error)
     }
 
+    updateValues(field: any, model: any) {
+        if(field.conditionalField && !isNull(model[field.conditionalField]) && field.clearOnConditionFail && !isConditionSatisfied(field, model)) {
+            this.modelChanged(null, () => {}, () => {})
```

Comment:
Why you are making it as empty.

Reply:
This is a conditional field. It is being rendered only if the condition evaluates to true.
If the value is already present it is not being removed when the condition evaluates to false.
If `clearOnConditionFail` is set to true, I am clearing the value.

Reply:
instead of this field, add `valueFn` field, which can evaluate what the value needs to be.